src/transaction/log_compress.h

```diff
@@ -55,10 +55,20 @@ struct log_zip
   LOG_ZIP_SIZE_T data_length;	/* length of stored (compressed/uncompressed)log_zip data */
   LOG_ZIP_SIZE_T buf_size;	/* size of log_zip data buffer */
   char *log_data;		/* compressed/uncompressed log_zip data (used as data buffer) */
+
+    log_zip ():data_length (0), buf_size (0), log_data (nullptr)
+  {
+  }
+  log_zip (const log_zip &) = delete;
+  log_zip (log_zip &&) = delete;
+
+  log_zip & operator= (const log_zip &) = delete;
+  log_zip & operator= (log_zip &&) = delete;
```

Comment:
For default initialization, the best option is to initialize members directly :).

Being a .h file, indent needs to be disabled.
```suggestion
  LOG_ZIP_SIZE_T data_length = 0;	/* length of stored (compressed/uncompressed)log_zip data */
  LOG_ZIP_SIZE_T buf_size = 0;	/* size of log_zip data buffer */
  char *log_data = nullptr;		/* compressed/uncompressed log_zip data (used as data buffer) */

  // *INDENT-OFF*
  log_zip () = default
  log_zip (const log_zip &) = delete;
  log_zip (log_zip &&) = delete;

  log_zip & operator= (const log_zip &) = delete;
  log_zip & operator= (log_zip &&) = delete;
  // *INDENT-ON*
```

Reply:
Done